src/Optical/ParaxialAnalysis/HeadEyeModel/Misc.jl

```diff
@@ -0,0 +1,76 @@
+
+
+# based on code from https://www.ilikebigbits.com/2015_03_04_plane_from_points.html
+"""
+    plane_from_points(points::Vector{SVector{3, Float64}}) ->  centroid, normal
+
+Estimate the best fitting plane for a set of points in 3D    
+"""
+function plane_from_points(points::Vector{SVector{3, Float64}}) 
+    if length(points) < 3 
+        return nothing      # At least three points required
+    end
+
+    sum = Base.sum(points)
+    centroid = Vec3(sum * (1.0 / (length(points))))
+
+    # Calc full 3x3 covariance matrix, excluding symmetries:
+    xx = 0.0 
+    xy = 0.0
+    xz = 0.0
+    yy = 0.0 
+    yz = 0.0
+    zz = 0.0
+
+    for p in points 
+        r = p - centroid;
+        xx += r[1] * r[1]
+        xy += r[1] * r[2]
+        xz += r[1] * r[3]
+        yy += r[2] * r[2]
+        yz += r[2] * r[3]
+        zz += r[3] * r[3]
+    end
+
+    det_x = yy*zz - yz*yz
+    det_y = xx*zz - xz*xz
+    det_z = xx*yy - xy*xy
+
+    det_max = max(det_x, det_y, det_z)
+    if det_max <= 0.0 
+        return nothing      # The points don't span a plane
+    end
+
+    # Pick path with best conditioning:
+    if      (det_max == det_x)
+        dir = Vec3(det_x,xz*yz - xy*zz,xy*yz - xz*yy)
+    elseif  (det_max == det_y)
+        dir = Vec3(xz*yz - xy*zz, det_y, xy*xz - yz*xx)
+    else
+        dir = Vec3(xy*yz - xz*yy,xy*xz - yz*xx,det_z)
+    end
```

Comment:
Is this computation time sensitive? If not then it might be simpler to use the SVD on a matrix of the points and use the singular vectors as the coordinate frame of the best fitting plane. This would only be a few lines of code:

y= x .- mean(x,dims=2)
U,_,_ = svd(y)

then choose which of the U vectors you want to align with the x,y,z axes. Or just return the largest singular vector. 

Built in svd on 3x100 Matrix takes 10usec. If you use MMatrix instead it takes 6us. Or you can use my JacobiSVD (available on request) on MMatrix and it takes 3usec.

Reply:
i'm not sure. i had some stability issues in the past with SVG in extreme cases - not sure if because of numerical errors or other reason. will check it out.

Reply:
@galran if you have stability issues then try my JacobiSVD code. It is supposed to compute small singular vectors to higher accuracy than the more common algorithms. And it is twice as fast as the Julia built in SVD.